docs/sources/flow/reference/components/loki.source.api.md

```diff
@@ -0,0 +1,82 @@
+Grafana Agents to send log entries over the network to other Grafana Agents.
+
+[loki.write]: {{< relref "./loki.write.md" >}}
+[loki-push-api]: https://grafana.com/docs/loki/latest/api/#push-log-entries-to-loki
+
+## Usage
+
+```river
+loki.source.api LABEL {
+    http_address = "LISTEN_ADDRESS"
+    http_port = PORT
+    forward_to = RECEIVER_LIST
+}
+```
+
+## Arguments
+
+`loki.source.api` supports the following arguments:
+
+ Name                     | Type                 | Description                                                | Default | Required 
+--------------------------|----------------------|------------------------------------------------------------|---------|----------
+ `http_port`              | `int`                | The host port for the HTTP server to listen on.            | `0`     | yes      
+ `http_address`           | `string`             | The host address for the HTTP server to listen on.         | `""`    | yes      
+ `forward_to`             | `list(LogsReceiver)` | List of receivers to send log entries to.                  |         | yes      
+ `use_incoming_timestamp` | `bool`               | Whether or not to use the timestamp received from request. | `false` | no       
+ `labels`                 | `map(string)`        | The labels to associate with each received logs record.    | `{}`    | no       
+ `relabel_rules`          | `RelabelRules`       | Relabeling rules to apply on log entries.                  | `{}`    | no       
+
+The `relabel_rules` field can make use of the `rules` export value from a
+[`loki.relabel`][loki.relabel] component to apply one or more relabeling rules to log entries
+before they're forwarded to the list of receivers in `forward_to`.
+
+[loki.relabel]: {{< relref "./loki.relabel.md" >}}
+
+## Exported fields
+
+`loki.source.api` does not export any fields.
+
+## Component health
+
+`loki.source.api` is only reported as unhealthy if given an invalid configuration.
+
+## Metrics
+
+The following are some of the metrics that are exposed when this component is used. Note that the metrics include labels
+such as `status_code` where relevant, which can be used to measure request success rates.
+
+* `loki_source_api_request_duration_seconds` - Time (in seconds) spent serving HTTP requests (histogram).
+* `loki_source_api_request_message_bytes` - Size (in bytes) of messages received in the request (histogram).
+* `loki_source_api_response_message_bytes` - Size (in bytes) of messages sent in response (histogram).
+* `loki_source_api_tcp_connections` - Current number of accepted TCP connections (gauge).
+
+## Example
+
+This example starts an HTTP server on `localhost` address and port `9999`. The server receives log entries and forwards
+them to a `loki.echo` component while
+adding a `forwarded="true"` label.
+
+```river
+loki.echo "print" {}
+
+loki.source.api "loki_push_api" {
+    http_address = "localhost"
```

Comment:
Since people will be copying this example, should we make the listen address here `0.0.0.0` so the component can also receive requests from other machines over the network? (since I guess this will be its primary use case)

Reply:
Good idea!